linode_api4/linode_client.py

```diff
@@ -1433,6 +1433,620 @@ def cancel(self):
+
+        :param acl: The Access Control Level of the bucket using a canned ACL string.
+                    For more fine-grained control of ACLs, use the S3 API directly.
+        :type acl: str
+                    Enum: private,public-read,authenticated-read,public-read-write
+
+        :param cluster: The ID of the Object Storage Cluster where this bucket
+                        should be created.
+        :type cluster: str
+
+        :param cors_enabled: If true, the bucket will be created with CORS enabled for
+                             all origins. For more fine-grained controls of CORS, use
+                             the S3 API directly.
+        :type cors_enabled: boolean
+
+        :param label: The name for this bucket. Must be unique in the cluster you are
+                      creating the bucket in, or an error will be returned. Labels will
+                      be reserved only for the cluster that active buckets are created
+                      and stored in. If you want to reserve this bucket’s label in
+                      another cluster, you must create a new bucket with the same label
+                      in the new cluster.
+        :type label: str
+
+        :returns: A Object Storage Buckets that created by user.
+        :rtype: ObjectStorageBucket
+        """
+        if acl not in (
+            "private",
+            "public-read",
+            "authenticated-read",
+            "public-read-write",
+        ):
+            raise ValueError("Invalid ACL value: {}".format(acl))
+
+        params = {
+            "cluster": cluster,
+            "label": label,
+            "acl": acl,
+            "cors_enabled": cors_enabled,
+        }
+
+        result = self.client.post("/object-storage/buckets", data=params)
+
+        if not "label" in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when creating Object Storage Bucket!",
+                json=result,
+            )
+
+        return ObjectStorageBucket(self.client, result["label"], result)
+
+    def buckets_in_cluster(self, cluster_id, *filters):
+        """
+        Returns a list of Buckets in this cluster belonging to this Account.
+
+        This endpoint is available for convenience.
+        It is recommended that instead you use the more fully-featured S3 API directly.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-buckets-in-cluster-list
+
+        :param cluster_id: The ID of the cluster this bucket exists in.
+        :type cluster_id: str
+
+        :returns: A list of Object Storage Buckets that in the requested cluster.
+        :rtype: PaginatedList of ObjectStorageBucket
+        """
+
+        return self.client._get_and_filter(
+            ObjectStorageBucket,
+            *filters,
+            endpoint="/object-storage/buckets/{}".format(cluster_id),
+        )
+
+    def bucket_delete(self, cluster_id, bucket):
```

Comment:
I think bucket deletion functionality is implicitly implemented for objects derived from `Base`, so we should be good to drop this one 👍

Reply:
I definitely think there's value in convenience wrappers like this so it's worth discussing ^

For existing resources, the flow for deleting a resource looks something like this:

```python
Instance(client, 12345).delete()
```

or using `client.load(...)`, which makes an extra GET request:

```python
client.load(Instance, 12345).delete()
```